plugins/music.py

```diff
@@ -509,33 +779,59 @@ class MusicPlayer(BasePlugin):
             await respond(self.client, data, "**ANALYSIS: Queue complete.**")
 
     async def add_song(self, vid, data):
-        before_args = " -reconnect 1 -reconnect_streamed 1 -reconnect_delay_max 30"
+        """
+        Adds songs to queue, no question asked
+        :param vid: URL or search query
+        :param data: message data for responding
+        :return:
+        """
+        before_args = "" if self.plugin_config["download_songs"] else " -reconnect 1 -reconnect_streamed 1 " \
+                                                                      "-reconnect_delay_max 30"
         t_loop = asyncio.get_event_loop()
         try:
-            t_player = await self.vc.create_ytdl_player(vid, ytdl_options=self.ytdl_options,
-                                                        before_options=before_args,
-                                                        after=lambda: t_loop.create_task(
-                                                            self.play_next(data)))
+            t_players, t_playlist = await self.create_player(vid, ytdl_options=self.ytdl_options,
+                                                             before_options=before_args,
+                                                             after=lambda: t_loop.create_task(self.play_next(data)))
         except DownloadError:
             await respond(self.client, data, "**NEGATIVE. Could not load song.**")
             return
-        self.logger.info(f"Adding {t_player.title} to music queue.")
-        self.queue.append(t_player)
+        for t_player in t_players:
+            t_player.author = data.author.id
+            self.logger.info(f"Adding {t_player.title} to music queue. Submitted by {t_player.author}.")
+            self.queue.append(t_player)
 
     def start_timer(self, loop):
         asyncio.set_event_loop(loop)
         try:
             loop.run_until_complete(self.display_time())
-        except:
-            pass
+        except Exception as e:
+            self.logger.error(f"ERROR {e}")
```

Comment:
Use `self.logger.exception("Error starting timer. ", exc_info=True)`; always include what function is erroring in an error message.